odc/stats/tasks.py

```diff
@@ -130,10 +129,23 @@ def __init__(
     def out_path(self, suffix: str) -> str:
         return out_path(suffix, self._output)
 
+    def ds_align(self, dss: Iterable, product: DatasetType, pair_size: int):
+
+        def pair_dss(groups, pair_size):
+            for _, ds_group in groups:
+                ds_group = tuple(ds_group)
+                if len(ds_group) == pair_size:
+                    yield ds_group
+        paired_dss = groupby(dss, key=lambda ds: (ds.center_time, ds.metadata.region_code))
```

Comment:
I'm pretty sure `center_time` always exists but `ds.metadata.region_code` is MetadataType specific.

Not sure a better approach, but it might be worth checking for the existence of a region_code and handling if it doesn't exist.

Reply:
done